src/v/kafka/server/group_manager.h

```diff
@@ -284,8 +284,20 @@ namespace kafka {
  * This batch consumer is used during partition recovery to read, index, and
  * deduplicate both group and commit metadata snapshots.
  */
+struct recovery_batch_consumer_state {
```

Comment:
this is great, but when we add or remove a `kafka::group_id` i think we should call `shrinkToFit()` or `rehash(0)`
as the group_ids will not change much and we can just run at `capacity()` of the map itself?